### DISCUSSION TITLE
Improve Developer Experience

### DISCUSSION THREAD

**Discussion by error-four-o-four** [michaelkolesidis/javascript-software-synthesizer#30] [Ideas]
Hey there,

I'd like to suggest some aspects to improve the DX:
- At first glance, the main entry file `jss.ts` is a bit overwhelming. Maybe one could split it into several modules (e.g. "renderer", "handlers", "constants", "synths" etc.).
- Add type checking and completion by either asserting the types or adding the "noImplicitAny"-rule to tsconfig.json

**Comment by michaelkolesidis** [comment]
Thank for your input!

I totally agree with the proposed impovements, feel free to work on them and submit a pull request.

**Comment by error-four-o-four** [comment]
Would you be opposed to add npm packages like prettier and lint-staged?? But one would have to define the prettier rules. So here's a very fundamental question: 'Tabs' or 'Spaces' ?? 😅
this is my default configuration:
```js
{
	"semi": true,
	"useTabs": true,
	"tabWidth": 2,
	"trailingComma": "es5",
	"arrowParens": "always",
	"bracketSameLine": true,
	"singleQuote": true,
	"endOfLine": "crlf"
}
```

**Comment by michaelkolesidis** [comment]
Feel free to add them, fine by me!
